lib/boavizta.ts

```diff
@@ -0,0 +1,135 @@
+import {IImpactModelInterface, IStaticParams} from "./interfaces/index";
+import axios from "axios";
+
+export {
+    IImpactModelInterface,
+    IStaticParams
+} from "./interfaces/index";
+
+
+export interface IBoaviztaStaticParams extends IStaticParams {
+    provider?: string;
+    componentType?: string;
+}
+
+export interface IBoaviztaCloudTelemetry {
+
+}
+
+
+export interface IBoaviztaAuthCredentials {
+}
+
+export class BoaviztaCloudImpactModel implements IImpactModelInterface {
+    public provider: string | undefined;
+    public name: string | undefined;
+    protected authCredentials: IBoaviztaAuthCredentials | undefined;
+
+    create(name: string, staticParams: object | undefined): object {
+        const staticParamCast = staticParams as IBoaviztaStaticParams;
+        if (staticParamCast?.hasOwnProperty('provider')) {
```

Comment:
Is there a good reason you are using hasOnwPropery instead of `in`, 

if ("provider" in statisParams) is much more succint.